artemis/base_pytest.py

```diff
@@ -236,40 +243,37 @@ def wait_for_rt_reload(self, last_rt_data_loaded, cov):
             raise Exception("real time data not loaded")
         logger.info('RT data reloaded at {}'.format(rt_data_loaded))
 
-    def request_compare(self, url):
+    def request_compare(self, http_query, checker):
         self.nb_call_to_request_compare += 1
 
-        # creating the url
-        http_query = config['URL_JORMUN'] + '/v1/coverage/' + str(self.data_sets[0]) + '/' + url
-
         # Get the json answer of the request (it is just a string here)
         http_response = requests.get(http_query)
 
-
         response_string = http_response.text
 
         if self.create_ref:
             # Create the reference file
-            self.create_reference(http_query, response_string)
+            self.create_reference(http_query, response_string, checker)
         else:
             # Comparing my response and my reference
-            self.compare_with_ref(http_query, response_string)
+            self.compare_with_ref(http_query, response_string, checker)
 
     def api(self, url, response_checker=default_checker.default_checker):
         """
-        used to check misc API
-
+        Call to an endpoint using a coverage
         NOTE: works only when one region is loaded for the moment (when needed change this)
         """
-        return self._api_call(url, response_checker)
+        coverage_query = "/coverage/{coverage_name}/{query}".format(coverage_name=str(self.data_sets[0]), query=url)
+        return self._api_call(coverage_query, response_checker)
 
     def _api_call(self, url, response_checker):
         """
         call the api and check against previous results
 
         the query is written in a file
         """
-        self.request_compare(url)
+        http_query = "{base_query}/v1{url}".format(base_query=config['URL_JORMUN'], url=url)
```

Comment:
minor, to stay consistent with other calls:
```suggestion
        http_query = "{base_query}/v1/{url}".format(base_query=config['URL_JORMUN'], url=url)
```
(then you can remove the additional `/` when calling this method)

Reply:
This won't work because of this test:
https://github.com/CanalTP/artemis/blob/7cb971c0bf6d04ba5443ce5b102edd8ff44f0e8f/artemis/tests/bibus_test.py#L120
A request with `//` will return an error